### PR TITLE
feat: add support for picklist deployment

### PR DESCRIPTION
**Pull request by gnemiq**
#### Checklist
All items have to be completed before a PR is merged

- [x] Adhere to [Contribution Guidelines](https://docs.dxatscale.io/about-us/contributing-to-dx-scale)
- [ ] Updates to Decision Records considered?
- [ ] Updates to documentation at [DX@Scale Guide](https://github.com/dxatscale/dxatscale-guide) considered?
- [ ] Tested changes?
- [ ] Unit Tests new and existing passing locally?

### CODE REVIEW

File: packages/core/src/package/preDeployer/PicklistEnabler.ts [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+    public async execute(
+        componentSet: ComponentSet,
+        conn: Connection,
+        logger: Logger
+    ) {
+
+        try {
+            let sourceComponents = componentSet.getSourceComponents().toArray();
+            let components = [];
```
Comment (azlam-abdulsalam) [inline packages/core/src/package/preDeployer/PicklistEnabler.ts]: @MengAQi  I think this should be optimized, as the during the build phase we already know the components, there is no need to analyze yet again

**Review by zhebinliu — APPROVED** [review]
LGTM

**Review by zhebinliu — APPROVED** [review]
LGTM

**Review by azlam-abdulsalam — APPROVED** [review]
Thanks @Rocko1204  @MengAQi  and @zhebinliu  for testing this one

**Pull request merged**